examples/pennylane_run_quantum_natural_gradient.py

```diff
@@ -0,0 +1,356 @@
+
+dev = qml.device('default.qubit', wires=3)
+
+@qml.qnode(dev)
+def circuit(params):
+    # non-parametrized gates
+    qml.RY(np.pi/4, wires=0)
+    qml.RY(np.pi/3, wires=1)
+    qml.RY(np.pi/7, wires=2)
+
+    # Parametrized layer 1
+    qml.RZ(params[0], wires=0)
+    qml.RZ(params[1], wires=1)
+
+    # non-parametrized gates
+    qml.CNOT(wires=[0, 1])
+    qml.CNOT(wires=[1, 2])
+
+    # Parametrized layer 2
+    qml.RY(params[2], wires=1)
+    qml.RX(params[3], wires=2)
+
+    # non-parametrized gates
+    qml.CNOT(wires=[0, 1])
+    qml.CNOT(wires=[1, 2])
+
+    return qml.expval(qml.PauliY(0))
+
+params = np.array([0.432, -0.123, 0.543, 0.233])
+
+##############################################################################
+# The above circuit consists of 4 parameters, with two distinct parametrized
+# layers of 2 parameters each. Therefore, the block-diagonal approximation
+# consists of two :math:`2\times 2` matrices, :math:`g^{(1)}` and :math:`g^{(2)}`.
+#
+# Computing the first block-diagonal :math:`g^{(1)}`, we create subcircuits consisting
+# of all gates prior to the layer, and observables corresponding to
+# the *generators* of the gates in the layer:
+
+g1 = np.zeros([2, 2])
+
+def layer1_subcircuit(params):
+    # all preceding gates
+    qml.RY(np.pi/4, wires=0)
+    qml.RY(np.pi/3, wires=1)
+    qml.RY(np.pi/7, wires=2)
+
+##############################################################################
+# The following subcircuit calculates the diagonal terms of :math:`g^{(1)}`,
+#
+# .. math::
+#
+#     g^{(1)}_{ii} = \langle \psi_\ell | K_i^2 | \psi_\ell\rangle
+#     - \langle \psi_\ell | K_i | \psi_\ell \rangle^2 = \Delta K_i
+
+@qml.qnode(dev)
+def layer1_diag(params):
+    layer1_subcircuit(params)
+    return var(qml.PauliZ(0)), var(qml.PauliZ(1))
+
+# calculate the diagonal terms
+varK0, varK1 = layer1_diag(params)
+g1[0, 0] = varK0/4
+g1[1, 1] = varK1/4
+
+##############################################################################
+# The following two subcircuits calculate the off-diagonal terms of :math:`g^{(1)}`:
+
+@qml.qnode(dev)
+def layer1_off_diag_single(params):
+    layer1_subcircuit(params)
+    return expval(qml.PauliZ(0)), expval(qml.PauliZ(1))
+
+@qml.qnode(dev)
+def layer1_off_diag_double(params):
+    layer1_subcircuit(params)
+    ZZ = np.kron(np.diag([1, -1]), np.diag([1, -1]))
+    return expval(qml.Hermitian(ZZ, wires=[0, 1]))
+
+# calculate the off-diagonal terms
+exK0, exK1 = layer1_off_diag_single(params)
+exK0K1 = layer1_off_diag_double(params)
+
+g1[0, 1] = (exK0K1 - exK0*exK1)/4
+g1[1, 0] = (exK0K1 - exK0*exK1)/4
+
+##############################################################################
+# Note that, by definition, the block-diagonal matrices must be real and
+# symmetric.
+#
+# We can repeat the above process to compute :math:`g^{(2)}`:
+
+g2 = np.zeros([2, 2])
+
+def layer2_subcircuit(params):
+    # all preceding gates
+    # non-parametrized gates

```

Comment:
the back-to-back comments here are a bit confusing. it feels like a typo or there is something missing

Reply:
I've hopefully fixed this, by adding a module docstring to clearly specify the purpose of the function, separately from the single-line comments